great_expectations/experimental/datasources/file_path_data_asset.py

```diff
@@ -158,11 +160,11 @@ def get_batch_list_from_batch_request(
             self.datasource.get_execution_engine()
         )
 
-        self._insure_single_data_connector_instance_exists()
-
         batch_definition_list: List[
             BatchDefinition
-        ] = self._data_connector.get_batch_definition_list(batch_request=batch_request)
+        ] = self._build_or_get_data_connector().get_batch_definition_list(
```

Comment:
This is fine, but given we are calling this methods multiple times within the same scope, I'd typically store the result in a local variable:
`data_connector = _build_or_get_data_connector()` and then later
`data_connector.get_batch_definition_list(..)` and 
`data_connector.build_batch_spec(...)`